docs/architecture/contract-components.md

```
@@ -0,0 +1,176 @@
+function removeFromBlacklistBatch(address[] calldata accounts) public onlyOwner
+function isBlacklisted(address account) public view returns (bool)
+function destroyBlackFunds(address blackListedUser) public onlyOwner
+```
+
+## Events
+
+The contract emits events for transparency and monitoring purposes. All events are indexed for efficient filtering.
+
+### Core Events
+
+**Transfer**
```

markdownlint-cli2 (0.18.1)

Emphasis used instead of a heading

(MD036, no-emphasis-as-heading)